File: wtplayer/src/WTAudioClip3D.cpp
#include "basetypes.hpp"

#include "WTAudioClip3D.hpp"
#include "WTCamera.hpp"

WTAudioClip3D::WTAudioClip3D(WT* wt_,
                             char* File_Name,
                             int WTCache_Type) :
  WTGroup(wt_) {
}

WTAudioClip3D::~WTAudioClip3D() {
  if(this->camera)
    this->camera->Release();
  this->camera = NULL;
}

void WTAudioClip3D::start(int loop,
                          int wait){
  APILOG;
}

void WTAudioClip3D::stop(){
  APILOG;
}

bool WTAudioClip3D::isPlaying(){
  APILOG;
  return false;
}

float WTAudioClip3D::getDoppler(){
  APILOG;
  return 0;
}

void WTAudioClip3D::setDoppler(float Rate){
  APILOG;
}

int WTAudioClip3D::getVolumeScale(){
  APILOG;
  return 0;
}

void WTAudioClip3D::setVolumeScale(int vol){
  APILOG;
}

float WTAudioClip3D::getPlaybackRateScale(){
  APILOG;
  return 0;
}

void WTAudioClip3D::setPlaybackRateScale(float Rate){
  APILOG;
}

float WTAudioClip3D::getMinDistance(){
  APILOG;
  return 0;
}

void WTAudioClip3D::setMinDistance(float min){
  APILOG;
}

float WTAudioClip3D::getMaxDistance(){
  APILOG;
  return 0;
}

void WTAudioClip3D::setMaxDistance(float max){
  APILOG;
}

int WTAudioClip3D::getPlaybackLength(){
  APILOG;
  return 0;
}

int WTAudioClip3D::getPlaybackPosition(){
  APILOG;
  return 0;
}

WTCamera* WTAudioClip3D::getCamera(){
  APILOG;
  return this->camera;
}

void WTAudioClip3D::setCamera(WTCamera* camera){
  APILOG;
  if(this->camera)
    this->camera->Release();
  this->camera = camera;
  this->camera->AddRef();
}

void WTAudioClip3D::removeCamera(){
  APILOG;
  if(this->camera) {
    this->camera->Release();
    this->camera = NULL;
  }
}

File: wtplayer/javasrc/wildtangent/webdriver/WTAudioClip3D.java
package wildtangent.webdriver;

import wildtangent.webdriver.WTCamera;
import wildtangent.webdriver.WTGroup;

public interface WTAudioClip3D
extends WTGroup {
    public void setPlaybackRateScale(float var1);

    public void stop();

    public float getMinDistance();

    public void setMinDistance(float var1);

    public void removeCamera();

    public int getVolumeScale();

    public void setVolumeScale(int var1);

    public float getDoppler();

    public void setDoppler(float var1);

    public float getMaxDistance();

    public void setMaxDistance(float var1);

    public int getPlaybackPosition();

    public boolean isPlaying();

    public int getPlaybackLength();

    public void start(int var1, int var2);

    public void start(int var1);

    public void start();

    public WTCamera getCamera();

    public void setCamera(WTCamera var1);

    public void setCamera();

    public float getPlaybackRateScale();
}

